### PR TITLE
Update contributing guide

### PR DESCRIPTION
**Pull request by jneira**
---
Please include the following checklist in your PR:

* [X] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [ ] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [X] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -240,15 +240,16 @@ There are a few main venues of communication:
 * For more organizational concerns, the [mailing
   list](http://www.haskell.org/mailman/listinfo/cabal-devel) is used.
 
-* Many developers idle on `#hackage` on `irc.freenode.net` ([archives](http://ircbrowse.net/browse/hackage)).  `#ghc` ([archives](http://ircbrowse.net/browse/ghc)) is also a decently good bet.
+* Many developers idle on `#hackage` on [irc.libera.net](https://libera.chat) ([archives](http://ircbrowse.net/browse/hackage)).  `#ghc` ([archives](http://ircbrowse.net/browse/ghc)) is also a decently good bet.
```
Comment (jneira) [inline CONTRIBUTING.md:239]: archives links are broken and i guess it should be updated to point to actual ones, if they even exists

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -240,15 +236,16 @@ There are a few main venues of communication:
...
+  * You can join the channel using a web client, even anonymously: https://kiwiirc.com/nextclient/irc.libera.chat/?nick=Guest%7C?#hackage
 
 Releases
 --------
 
 Notes for how to make a release are at the
 wiki page ["Making a release"](https://github.com/haskell/cabal/wiki/Making-a-release).
-Currently, @23Skidoo, @rthomas, @tibbe and @dcoutts have access to
-`haskell.org/cabal`, and @davean is the point of contact for getting
+Currently, [@emilypi](https://github.com/emilypi) and [@gbaz](https://github.com/gbaz) have access to
```
Comment (jneira) [inline CONTRIBUTING.md:247]: not sure at all about the chosen maintainers here

//cc @emilypi @gbaz @Mikolaj

**Comment by jneira** [comment]
api docs mentioned here: https://github.com/haskell/cabal/pull/7848/files#L257 are for cabal-3.3.0.0 and no master:

http://haskell.github.io/cabal-website/doc/html/Cabal/

File: CONTRIBUTING.md (hunk 1 above)
Comment (fgaz) [reply CONTRIBUTING.md:239]: We could use matrix for that (if we do want archives) https://view.matrix.org/room/%21VeTpfwcnSnUZwjyccf:libera.chat/

Also s/libera.net/libera.chat/

File: CONTRIBUTING.md (hunk 2 above)
Comment (fgaz) [reply CONTRIBUTING.md:247]: Nevermind, I misread haskell.org/cabal as github.com/cabal

File: CONTRIBUTING.md (hunk 1 above)
Comment (jneira) [reply CONTRIBUTING.md:239]:
```suggestion
* Many developers idle on `#hackage` on https://libera.chat ([archives](http://ircbrowse.net/browse/hackage)).  `#ghc` ([archives](http://ircbrowse.net/browse/ghc)) is also a decently good bet.
```

Comment (jneira) [reply CONTRIBUTING.md]: > We could use matrix for that (if we do want archives) https://view.matrix.org/room/%21VeTpfwcnSnUZwjyccf:libera.chat/

It gives to me:

```
User @2985302:matrix.org not in room !VeTpfwcnSnUZwjyccf:libera.chat, and room previews are disabled (M_FORBIDDEN)
```

**Review by andreasabel — APPROVED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -240,15 +236,17 @@ There are a few main venues of communication:
 * For more organizational concerns, the [mailing
   list](http://www.haskell.org/mailman/listinfo/cabal-devel) is used.
 
-* Many developers idle on `#hackage` on `irc.freenode.net` ([archives](http://ircbrowse.net/browse/hackage)).  `#ghc` ([archives](http://ircbrowse.net/browse/ghc)) is also a decently good bet.
+* Many developers idle on `#hackage` on https://libera.chat.  The `#ghc` channel is also a decently good bet.
```
Comment (jneira) [inline CONTRIBUTING.md:239]:
```suggestion
* Many developers idle on `#hackage` on https://libera.chat. The `#ghc` channel is also a decently good bet.
```

**Comment by jneira** [comment]
@fgaz hi!, would be this good to go now?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -240,15 +236,17 @@ There are a few main venues of communication:
 * For more organizational concerns, the [mailing
   list](http://www.haskell.org/mailman/listinfo/cabal-devel) is used.
 
-* Many developers idle on `#hackage` on `irc.freenode.net` ([archives](http://ircbrowse.net/browse/hackage)).  `#ghc` ([archives](http://ircbrowse.net/browse/ghc)) is also a decently good bet.
+* Many developers idle on `#hackage` on https://libera.chat. The `#ghc` channel is also a decently good bet.
+  * You can join the channel using a web client, even anonymously: https://kiwiirc.com/nextclient/irc.libera.chat/?nick=Guest%7C?#hackage
```
Comment (fgaz) [inline CONTRIBUTING.md:240]: We should link to the [official webclient](https://web.libera.chat/#hackage) rather than kiwi.

**Review by fgaz — APPROVED** [review]
Looks good, minor comment

File: CONTRIBUTING.md (hunk 4 above)
Comment (jneira) [reply CONTRIBUTING.md:240]: yeah, will change

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -240,15 +236,17 @@ There are a few main venues of communication:
...
+  * Alternatively you can join it using [matrix](https://matrix.org/): https://matrix.to/#/#hackage:libera.chat
 
 Releases
 --------
 
 Notes for how to make a release are at the
 wiki page ["Making a release"](https://github.com/haskell/cabal/wiki/Making-a-release).
-Currently, @23Skidoo, @rthomas, @tibbe and @dcoutts have access to
-`haskell.org/cabal`, and @davean is the point of contact for getting
+Currently, [@emilypi](https://github.com/emilypi) and [@gbaz](https://github.com/gbaz) have access to
```
Comment (Mikolaj) [inline CONTRIBUTING.md:248]: @gbaz says the correct list is "is fgaz, tapas and mikolaj" (I was just added)

Comment (jneira) [inline CONTRIBUTING.md:248]:
```suggestion
Currently, [@emilypi](https://github.com/emilypi), [@fgaz](https://github.com/fgaz) and [@Mikolaj](https://github.com/Mikolaj) have access to
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -240,15 +236,17 @@ There are a few main venues of communication:
...
+`haskell.org/cabal`, and [@mikolaj](https://github.com/mikolaj) is the point of contact for getting
```
Comment (jneira) [inline CONTRIBUTING.md:249]:
```suggestion
`haskell.org/cabal`, and [@Mikolaj](https://github.com/Mikolaj) is the point of contact for getting
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -240,15 +236,17 @@ There are a few main venues of communication:
 * For more organizational concerns, the [mailing
   list](http://www.haskell.org/mailman/listinfo/cabal-devel) is used.
 
-* Many developers idle on `#hackage` on `irc.freenode.net` ([archives](http://ircbrowse.net/browse/hackage)).  `#ghc` ([archives](http://ircbrowse.net/browse/ghc)) is also a decently good bet.
+* Many developers idle on `#hackage` on https://libera.chat. The `#ghc` channel is also a decently good bet.
```
Comment (jneira) [inline CONTRIBUTING.md:239]:
```suggestion
* Many developers idle on `#hackage` on [`irc.libera.chat`](https://libera.chat). The `#ghc` channel is also a decently good bet.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -240,15 +236,17 @@ There are a few main venues of communication:
 * For more organizational concerns, the [mailing
   list](http://www.haskell.org/mailman/listinfo/cabal-devel) is used.
 
-* Many developers idle on `#hackage` on `irc.freenode.net` ([archives](http://ircbrowse.net/browse/hackage)).  `#ghc` ([archives](http://ircbrowse.net/browse/ghc)) is also a decently good bet.
+* Many developers idle on `#hackage` on [`irc.libera.chat`](https://libera.chat). The `#ghc` channel is also a decently good bet.
+  * You can join the channel using a web client, even anonymously: https://kiwiirc.com/nextclient/irc.libera.chat/?nick=Guest%7C?#hackage
```
Comment (jneira) [inline CONTRIBUTING.md:240]:
```suggestion
  * You can join the channel using a web client, even anonymously: https://web.libera.chat/#hackage
```

**Pull request merged**